e2e test breaks sometimes for TestGiteaParamsOnRepoCR

Issue:

```
{"level":"info","ts":1690892691.182491,"caller":"repository/teardown.go:16","msg":"Deleting NS pac-e2e-test-tlrhk"}
--- FAIL: TestGiteaParamsOnRepoCR (7.90s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 829 [running]:
testing.tRunner.func1.2({0x2cfba40, 0xc000ad9f20})
	/opt/hostedtoolcache/go/1.19.11/x64/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.19.11/x64/src/testing/testing.go:1399 +0x5f0
panic({0x2cfba40, 0xc000ad9f20})
	/opt/hostedtoolcache/go/1.19.11/x64/src/runtime/panic.go:890 +0x262
github.com/openshift-pipelines/pipelines-as-code/test.TestGiteaParamsOnRepoCR(0x0?)
	/home/runner/work/pipelines-as-code/pipelines-as-code/test/gitea_test.go:645 +0xf09
testing.tRunner(0xc000648340, 0x2f6c4a8)
	/opt/hostedtoolcache/go/1.19.11/x64/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.19.11/x64/src/testing/testing.go:1493 +0x75e
FAIL	github.com/openshift-pipelines/pipelines-as-code/test	421.175s
FAIL
make: *** [Makefile:63: test-e2e] Error 1
Error: Process completed with exit code 2.
```

--- a/test/gitea_test.go
+++ b/test/gitea_test.go
@@ -22,22 +22,21 @@ import (
 	tknpacgenerate "github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/generate"
 	tknpaclist "github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/list"
 	tknpacresolve "github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/resolve"
-	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
-	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/configmap"
-	pacrepo "github.com/openshift-pipelines/pipelines-as-code/test/pkg/repository"
-	"gopkg.in/yaml.v2"
-
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/git"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/settings"
 	tknpactest "github.com/openshift-pipelines/pipelines-as-code/test/pkg/cli"
+	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/configmap"
 	tgitea "github.com/openshift-pipelines/pipelines-as-code/test/pkg/gitea"
 	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/options"
 	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/payload"
+	pacrepo "github.com/openshift-pipelines/pipelines-as-code/test/pkg/repository"
 	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/secret"
 	twait "github.com/openshift-pipelines/pipelines-as-code/test/pkg/wait"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	"github.com/tektoncd/pipeline/pkg/names"
+	"gopkg.in/yaml.v2"
 	"gotest.tools/v3/assert"
 	"gotest.tools/v3/env"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
@@ -637,6 +636,8 @@ func TestGiteaParamsOnRepoCR(t *testing.T) {
 	repo, err := topts.ParamsRun.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(topts.TargetNS).Get(context.Background(), topts.TargetNS, metav1.GetOptions{})
 	assert.NilError(t, err)
 
+	// Checking repo status length to avoid index out of range [0] with length 0 issue
+	assert.Assert(t, len(repo.Status) != 0)
 	assert.NilError(t,
 		twait.RegexpMatchingInPodLog(context.Background(),
 			topts.ParamsRun,

